src/Filters/Has.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+use LaravelJsonApi\Eloquent\Schema;
+
+class Has implements Filter
+{
+    use Concerns\DeserializesValue;
+    use Concerns\IsSingular;
+
+    /**
+     * @var string
+     */
+    private string $relationName;
+
+    private Schema $schema;
+
+    /**
+     * SearchFilter constructor.
+     *
+     * @param string $relationName
+     * @param array $columns
+     */
+    public function __construct(string $relationName, Schema $schema)
+    {
+        $this->relationName = $relationName;
+        $this->schema = $schema;
+
+        if (!$this->schema->isRelationship($relationName)) {
+            throw new \LogicException("Relationship with name $relationName not defined in ". get_class($schema)." schema.");
+        }
+    }
+
+    /**
+     * Create a new filter.
+     *
+     * @param string $relationName
+     * @param Schema $schema
+     * @return WhereHas
+     */
+    public static function make(string $relationName, Schema $schema): self
```

Comment:
Make changes here to match changes to the constructor. Also, move the `make()` method above the constructor.